kore/app/parser/Main.hs

```diff
@@ -183,13 +198,32 @@ mainVerify
             ModuleName
             (VerifiedModule StepperAttributes)
         )
-mainVerify definition = flip runLoggerT Log.emptyLogger $ do
+mainVerify definition =
+    flip
+        runLoggerT
+        (Log.stderrLogger
+            (ExeName "kore-parser")
+            (fromNanoSecs 0)
+            TimestampsEnable
+        )
+        $ do
     verifyResult <-
         clockSomething "Verifying the definition"
             (verifyAndIndexDefinition
                 Builtin.koreVerifiers
                 definition
             )
+        & handle handleSomeException
     case verifyResult of
         Left err1            -> liftIO $ die $ printError err1
         Right indexedModules -> return indexedModules
+
+handleSomeException
+    :: (MonadThrow m, Log.MonadLog m)
+    => SomeException
+    -> m (Either (Error VerifyError) right)
+handleSomeException someException = do
+    case fromException someException of
+        Just (Log.SomeEntry entry) -> Log.logEntry entry
+        Nothing -> errorException someException
+    pure $ Left (koreError (displayException someException))
```

Comment:
What exactly is going on here?
Won't this line only be executed if the case above takes the `Just` branch? And if so, wouldn't it make more sense to just throw the error inside that branch?